### PR TITLE
Feature: Setup Association

### PR DESCRIPTION
**Pull request by Akorede3133**
## In this PR, the following were affected:

 * Created a class Classroom with the following:

    - [ ] @label instance variable,  initialized in the constructor.
    - [ ] Setter and getter for @label (remember about attr_accessor).

* Created the has-many/belongs-to relationship between Classroom and Student. The following was implemented:

    - [ ] Created the has-many side (a classroom has many students).
    - [ ] Created the belongs-to side (a student belongs to a classroom).
    - [ ] Made sure that when adding a student to a classroom it also sets the classroom for the student.
    - [ ] Made sure that when setting the classroom for a student it also adds it to the classrooms' students.

* Created a class Book with the following:

    - [ ] @title and @author instance variables, should be initialized in the constructor.
    - [ ] Setters and getters for instance variables (remember about attr_accessor).
  
* Created a class Rental with the following:

    - [ ] @date instance variable, should be initialized in the constructor.
    - [ ] Setter and getter for @date (remember about attr_accessor).

* Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following was implemented:
    - [ ] Created the has-many side of Book and Rental (a book has many rentals).
    - [ ] Created the belongs-to side of Rental and Book (a rental belongs to a book).
    - [ ] Created the has-many side of Person and Rental (a person has many rentals).
    - [ ] Created the belongs-to side of Rental and Person (a rental belongs to a person).
    - [ ] Modified the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    rental.book = self
+    @rentals.push(rental)
+  end
```
Comment (uwadonat) [inline book.rb]: `person` and `date` should also be added as permeters for new rental.
```
 def add_rental(person, date)
       Rental.new(date, self, person)
 end
end
```

File: person.rb [hunk 2]
```diff
@@ -2,20 +2,25 @@
...
 
   def can_use_services?
     of_age? || @parent_permission
   end
-
+  
+  def add_rental(rental)
+    rental.person = self
+    @rentals.push(rental)
+  end
```
Comment (uwadonat) [inline person.rb]: - [ ] Let's add `book` and `date`  as parameters for new rental.
```
def add_rental(book, date)
  rental.new(date, book, selt)
end
```

**Review by uwadonat — CHANGES_REQUESTED** [review]
## Hi @Akorede3133 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Your commit messages are very descriptive ✔️
- Organized and clean project ✔️
- Correct workflow was used ✔️

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (Akorede3133) [reply book.rb]: I noticed that you're not allowing the external creation of Rental instances. I assumed we could externally create a Rental instance with the necessary arguments.

Both approaches work perfectly.

**Review by shahadat3669 — APPROVED** [review]
Hi @Akorede3133 , 

Your project is complete!  There is nothing else to say other than it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ No linter Errors
✅ Followed previous reviewer instruction perfectly 👍
✅ The code is top-notch! 💯

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**